src/components/Formations/Formations.jsx

```diff
@@ -0,0 +1,185 @@
+import PropTypes from 'prop-types'
+import { Box, Container, Grid, IconButton, Typography } from '@mui/material'
+import SchoolRoundedIcon from '@mui/icons-material/SchoolRounded'
+import LinkIcon from '@mui/icons-material/Link'
+import InsertDriveFileIcon from '@mui/icons-material/InsertDriveFile'
+import DeleteIcon from '@mui/icons-material/Delete'
+import { styled } from '@mui/system'
+import InfiniteScroll from 'react-infinite-scroll-component'
+import { useInfiniteQueryWithScope } from 'api/useQueryWithScope'
+import { getNextPageParam, usePaginatedData } from 'api/pagination'
+import { getFormations } from 'api/formations'
+import { useErrorHandler } from 'components/shared/error/hooks'
+import PageHeader from 'ui/PageHeader'
+import { PageHeaderButton } from 'ui/PageHeader/PageHeader'
+import Loader from 'ui/Loader'
+import EmptyContent from 'ui/EmptyContent'
+import UICard from 'ui/Card/Card'
+import Button from 'ui/Button'
+
+const messages = {
+  title: 'Formations',
+  create: 'Ajouter',
+  edit: 'Modifier',
+  noFormation: "Vous n'avez aucune formation créée",
+  published: 'Publiée',
+  unpublished: 'Non publiée',
+  view: 'Afficher',
+  download: 'Télécharger',
+}
+
+const Badge = styled('span')(({ theme, isPublish = false }) => ({
+  ...(isPublish
+    ? {
+        backgroundColor: theme.palette.colors.blue[500],
+        color: theme.palette.colors.white,
+      }
+    : {
+        backgroundColor: '#fee2e2',
+        color: '#991b1b',
+      }),
+  borderRadius: '4px',
+  padding: '4px 8px',
+  fontSize: '12px',
+  fontWeight: '500',
+  marginRight: '8px',
+}))
+
+const Link = ({ href, children, ...props }) => (
+  <Typography component="a" href={href} sx={{ fontSize: '14px' }} {...props}>
+    {children}
+  </Typography>
+)
+
+Link.propTypes = {
+  href: PropTypes.string,
+  children: PropTypes.node,
+}
+
+const Formations = () => {
+  const { handleError } = useErrorHandler()
+  const {
+    data: paginatedFormations = null,
+    fetchNextPage,
+    hasNextPage,
+    isLoading,
+  } = useInfiniteQueryWithScope(
+    ['paginated-formations', { feature: 'Formations', view: 'Formations' }],
+    getFormations,
+    {
+      getNextPageParam,
+      onError: handleError,
+    }
+  )
+
+  const formations = usePaginatedData(paginatedFormations)
+
+  if (!formations || formations.length === 0)
+    <EmptyContent
+      description={messages.noFormation}
+      action={
+        <>
+          <PageHeaderButton label={messages.create} onClick={() => {}} icon={<SchoolRoundedIcon />} isMainButton />
+        </>
+      }
+    />
+
+  {
+    isLoading && <Loader />
+  }
```

Comment:
est-ce que les 2 versions sont identiques? si oui, pourquoi ne pas écrire avec IF? 
```suggestion
  if (isLoading) {
    return <Loader />
  }
```